language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java

```diff
@@ -59,8 +59,8 @@ public static List<LSCompletionItem> getRecordFieldCompletionItems(BallerinaComp
                                                                        RawTypeSymbolWrapper wrapper) {
         List<LSCompletionItem> completionItems = new ArrayList<>();
         fields.forEach((name, field) -> {
-            String insertText =
-                    getRecordFieldCompletionInsertText(field, 1);
+            DefaultValueGenerationUtil.SnippetContext snippetContext = new DefaultValueGenerationUtil.SnippetContext();
```

Comment:
Here it's correct to create snippet context per iteration since each iteration generates a separate insert text